webapp/app/common/calls.ts

```diff
@@ -0,0 +1,20 @@
+import { addressSchema, hexSchema } from "@/common/basic-schemas";
+import { numberToHex, parseEther } from "viem";
+import { z } from "zod";
+
+export const callSchema = z.object({
+  target: hexSchema,
+  data: hexSchema,
+  value: hexSchema,
+});
+
+export type Call = z.infer<typeof callSchema>;
+
+export const formCallSchema = z.object({
+  target: addressSchema,
+  data: hexSchema,
+  value: z
+    .string()
+    .refine((str) => !Number.isNaN(Number.parseFloat(str)), "Should be a valid number")
```

Comment:
```suggestion
    .coerce.number()
```

Reply:
I didn't use this from the beginning because I need to keep this as a string for the next line. But it's true that looks so much nicer than what I did. Let me apply this and make some adjustments to make the next line work.